redfish-core/lib/log_services.hpp

```diff
@@ -6874,32 +6954,22 @@ inline void fillSystemHardwareIsolationLogEntry(
                         messages::internalError(asyncResp->res);
                         break;
                     }
-
-                    if (*severity ==
-                        "xyz.openbmc_project.HardwareIsolation.Entry.Type.Critical")
-                    {
-                        entryJson["Severity"] = "Critical";
-                    }
-                    else if (
-                        *severity ==
-                        "xyz.openbmc_project.HardwareIsolation.Entry.Type.Warning")
-                    {
-                        entryJson["Severity"] = "Warning";
-                    }
-                    else if (
-                        *severity ==
-                        "xyz.openbmc_project.HardwareIsolation.Entry.Type.Manual")
+                    std::string severityString = *severity;
+                    std::string guardType;
+                    guardType =
```

Comment:
single line to avoid another copy constructor
std::string guardType = severityString.substr...

Reply:
done